src/app/ui/components/input/input.tsx

```diff
@@ -0,0 +1,215 @@
+      left: 'space.04',
+      zIndex: 9,
+      color: 'inherit',
+      textStyle: 'body.02',
+      transition: 'font-size 100ms ease-in-out, transform 100ms ease-in-out',
+      // Move the input's label to the top when the input has a value
+      '[data-has-value="true"] &': transformedLabelStyles,
+      '[data-shrink="true"] &': transformedLabelStyles,
+    },
+  },
+});
+
+type InputChldren = 'root' | 'label' | 'input';
+
+const { withProvider, withContext } = createStyleContext(input);
+
+interface InputContextProps {
+  hasValue: boolean;
+  setHasValue(hasValue: boolean): void;
+  registerChild(child: string): void;
+  children: InputChldren[];
+}
+
+const InputContext = createContext<InputContextProps | null>(null);
+
+function useInputContext() {
+  const context = useContext(InputContext);
+  if (!context) throw new Error('useInputContext must be used within an Input.Root');
+  return context;
+}
+
+const RootBase = withProvider('div', 'root');
+
+interface RootProps extends ComponentProps<'div'> {
+  hasError?: boolean;
+  /**
+   * Display the label in a top fixed position. Often necessary when
+   * programmatically updating inputs, similar to issues described here
+   * https://mui.com/material-ui/react-text-field/#limitations
+   */
+  shrink?: boolean;
+}
+function Root({ hasError, shrink, ...props }: RootProps) {
+  const [hasValue, setHasValue] = useState(false);
+  const [children, setChildren] = useState<InputChldren[]>(['root']);
+
+  function registerChild(child: InputChldren) {
+    setChildren(children => [...children, child]);
+  }
+
+  const dataAttrs = {
+    ...propIfDefined('data-has-error', hasError),
+    ...propIfDefined('data-shrink', shrink),
+    'data-has-label': children.includes('label'),
+  };
+
+  return (
+    <InputContext.Provider value={{ hasValue, setHasValue, children, registerChild }}>
+      <RootBase data-has-value={hasValue} {...dataAttrs} {...props} />
+    </InputContext.Provider>
+  );
+}
+
+const FieldBase = withContext('input', 'input');
+
+const Field = forwardRef(({ type, ...props }: ComponentProps<'input'>, ref) => {
+  const { setHasValue } = useInputContext();
+  const innerRef = useRef<HTMLInputElement | null>(null);
+
+  useImperativeHandle(ref, () => innerRef.current);
+
+  // We need to determine whether the input has a value on it's initial
+  // render. In many places we use Formik to apply default form values.
+  // Formik sets these values after the initial render, so we need to wait
+  // before doing this check to see if there's a value.
+  useOnMount(
+    () =>
+      void setTimeout(() => {
+        if (innerRef.current?.value !== '') setHasValue(true);
+      }, hackyDelayOneMs)
+  );
+
+  // `type=number` is bad UX, instead we follow guidance here
+  // https://mui.com/material-ui/react-text-field/#type-quot-number-quot
+  // https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/
+  const inputTypeProps =
```

Comment:
Maybe we an set `type=number` for `fee` 
![Screenshot 2024-01-24 at 14 25 55](https://github.com/leather-wallet/extension/assets/2938440/47a11471-c2d6-4b67-bbff-2ae304c280d1)

Reply:
investigating

Reply:
I don't think it's needed for this PR but good to solve